analyzers/src/SonarAnalyzer.CSharp/Rules/LoggerMembersNamesShouldComply.cs

```diff
@@ -0,0 +1,90 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Text.RegularExpressions;
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class LoggerMembersNamesShouldComply : ParametrizedDiagnosticAnalyzer
+{
+    private const string DiagnosticId = "S6669";
+    private const string MessageFormat = "Rename this {0} '{1}' to match the regular expression '{2}'.";
+    private const string DefaultFormat = "^_?[Ll]og(ger)?$";
+
+    private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat, isEnabledByDefault: false);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+        ImmutableArray.Create(Rule);
+
+    [RuleParameter("format", PropertyType.RegularExpression, "Regular expression used to check the field or property names against", DefaultFormat)]
+    public string Format { get; set; } = DefaultFormat;
+
+    private Regex NameRegex => new(Format, RegexOptions.Compiled, RegexConstants.DefaultTimeout);
+
+    private static readonly ImmutableArray<KnownType> Loggers = ImmutableArray.Create(
+        KnownType.Microsoft_Extensions_Logging_ILogger,
+        KnownType.Microsoft_Extensions_Logging_ILogger_TCategoryName,
+        KnownType.Serilog_ILogger,
+        KnownType.NLog_ILogger,
+        KnownType.NLog_ILoggerBase,
+        KnownType.NLog_Logger,
+        KnownType.log4net_ILog,
+        KnownType.log4net_Core_ILogger,
+        KnownType.Castle_Core_Logging_ILogger);
+
+    // Rename this [field|property] '[name]' to match the regular expression [format].
+    protected override void Initialize(SonarParametrizedAnalysisContext context) =>
+        context.RegisterNodeAction(c =>
+            {
+                foreach (var memberData in Declarations(c.Node))
+                {
+                    if (!NameRegex.SafeIsMatch(memberData.Name)
```

Comment:
Looking at the rules of MMF#2, I think we _really_ need to prioritize https://github.com/SonarSource/sonar-dotnet/issues/8851, since a lot of them need to check every single (field|invocation|argument), which will probably be more expensive than checking if a library is referenced.